### ISSUE TITLE
`require-param`: Linting destructured parameters with partial jsdoc leads to odd roots

### ISSUE THREAD

**Issue by underflowdev** [gajus/eslint-plugin-jsdoc#762]
Perhaps the answer here is "Don't do that".

## Expected behavior

Linting with
```js
  /**
   */
  quux({ foo }, { bar }) {
    console.log(foo, bar);
  }
```
results in
```js
  /**
   * @param root0
   * @param root0.foo
   * @param root1
   * @param root1.bar
   */
  quux({ foo }, { bar }) {
    console.log(foo, bar);
  }
```
All good.

## Actual behavior

Linting with
```js
  /**
   * @param root0
   * @param root0.foo
   */
  quux({ foo }, { bar }) {
    console.log(foo, bar);
  }
```
results in
```js
  /**
   * @param root0
   * @param root0.foo
   * @param root0.foo.bar
   */
  quux({ foo }, { bar }) {
    console.log(foo, bar);
  }
```
and accompanying warnings.

## ESLint Config
jsdoc/recommended

## ESLint sample

See above

## Environment

- Node version:  v16.0.0
- ESLint version  v6.8.0
- `eslint-plugin-jsdoc` version: 35.4.1

**Comment by brettz9** [comment]
This might be connected to #540 even while the details differ.
